shared/js/sync/errors.js

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+(function(exports) {
+
+  var unrecoverableErrors = {
+    ERROR_REQUEST_SYNC_REGISTRATION: 'Cannot register sync request',
+    ERROR_GET_FXA_ASSERTION: 'Cannot get Firefox Accounts assertion',
+    ERROR_SYNC_REQUEST: 'Cannot perform sync request'
+  };
+
+  var recoverableErrors = {
+    ERROR_SYNC_APP_KILLED:
+      'The app was killed while performaning a sync operation',
+    ERROR_SYNC_APP_GENERIC: 'Error while trying to sync'
```

Comment:
System doesn't need to contain any l10n string. It's up to the Settings app to translate these errors into user understandable errors. All that System will be giving is error codes.

Anyway, I am refactoring this code to remove the strings and convert them into simple error codes.